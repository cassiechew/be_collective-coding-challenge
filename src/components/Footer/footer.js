import './footer.scss';

/**
 * Component to hold a footer
 * 
 * @Component
 */
export default function Footer(props) {
    return(
        <div className="container__footer">
            <h3>Total Files: {props.numFiles}</h3>
            <h3>Total File Size: {props.fileSize} B</h3>
        </div>
    )
}