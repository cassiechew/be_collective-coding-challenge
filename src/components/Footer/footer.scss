div {
    .container__footer {
        padding-left: 15%;
    }
}