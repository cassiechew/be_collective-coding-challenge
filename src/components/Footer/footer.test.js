import React from 'react';
import Footer from './footer';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

describe('Footer component', () => {
    test('contains size text', () => {
        render(<Footer />);
        const totalText = screen.getByText(/Total Files:/i);
        expect(totalText).toBeInTheDocument();
    });

    test('contains file text', () => {
        render(<Footer />);
        const totalFileText = screen.getByText(/Total File Size:/i);
        expect(totalFileText).toBeInTheDocument();
    });
});

describe('snapshot test', () => {
    test('testing footer', () => {
      let tree = create(<Footer />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
});