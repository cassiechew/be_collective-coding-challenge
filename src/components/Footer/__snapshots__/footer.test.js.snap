// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test testing footer 1`] = `
<div
  className="container__footer"
>
  <h3>
    Total Files: 
  </h3>
  <h3>
    Total File Size: 
     B
  </h3>
</div>
`;
